new title, summary and headlines for proxy guide

DGuhr opened the pull request:
Closes #9570

DGuhr committed:
new title, summary and headlines for proxy guide

Closes #9570

stianst commented on docs/guides/src/main/server/proxy.adoc:
```diff
@@ -18,22 +19,22 @@ The `reencrypt` mode requires communication through HTTPS between the proxy and
+== Configure the proxy mode in Keycloak
+To select the proxy mode, run:
 
 <@kc.all parameters="--proxy <mode>"/>
```
I'm not super keen on @kc.all as that could be used by all "runtime" options, and this can easily lead into a bit of a mix/match. I'd rather just stick with a general rule of using `kc build` for build-time options, and `kc start` for runtime options. Then in how to configure guide we can cover that runtime options can also be included in config files, added to build command, etc.

stianst replied:
@kc.all would also be pretty hard to use if we decide to do the whole different versions of the docs for different environments.

stianst reviewed (commented):
LGTM just wanted to ask your opinion on removing `@kc.all` macro and using `@kc.start` instead.

DGuhr commented:
@stianst I removed the kc.all macro now, your comment made sense. Also removed unneeded "expectedValues" in the proxy guide and changed "Proxy modes overview" to "Available proxy modes". Pls re-review.

cc @pedroigor bc not sure if its used in another open PR. Will check the open TLS/Features docs PR from my side and change it accordingly ("build" for buildtime, "start" for runtime).
